docs/dotnet/dotnet-aspnet-metrics.md

```diff
@@ -0,0 +1,208 @@
+
+<!-- semconv metric.aspnet.rate_limiting.queued_requests(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `aspnet.rate_limiting.policy` | string | Rate limiting policy name. | `fixed`; `sliding`; `token` | Conditionally Required: [1] |
+
+**[1]:** if the matched endpoint for the request had a rate-limiting policy.
+<!-- endsemconv -->
+
+### Metric: `aspnet.rate_limiting.request.time_in_queue`
+
+<!-- semconv metric.aspnet.rate_limiting.request.time_in_queue(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `aspnet.rate_limiting.request.time_in_queue` | Histogram | `s` | The time request spent in a queue, waiting to acquire a rate limiting lease. [1] |
+
+**[1]:** Meter name is `Microsoft.AspNetCore.RateLimiting`
+<!-- endsemconv -->
+
+<!-- semconv metric.aspnet.rate_limiting.request.time_in_queue(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `aspnet.rate_limiting.policy` | string | Rate limiting policy name. | `fixed`; `sliding`; `token` | Conditionally Required: [1] |
+| `aspnet.rate_limiting.result` | string | Rate-limiting result, shows whether lease was acquired or contains rejection reason | `acquired`; `request_canceled` | Required |
+
+**[1]:** if the matched endpoint for the request had a rate-limiting policy.
+
+`aspnet.rate_limiting.result` has the following list of well-known values. If one of them applies, then the respective value MUST be used, otherwise a custom value MAY be used.
+
+| Value  | Description |
+|---|---|
+| `acquired` | lease acquired |
+| `endpoint_limiter` | Lease request was rejected by the endpoint limiter |
+| `global_limiter` | Lease request was rejected by the global limiter |
+| `request_canceled` | Lease request was canceled |
+<!-- endsemconv -->
+
+### Metric: `aspnet.rate_limiting.requests`
+
+<!-- semconv metric.aspnet.rate_limiting.requests(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `aspnet.rate_limiting.requests` | Counter | `{request}` | Number of requests that tried to acquire a rate limiting lease. [1] |
+
+**[1]:** Requests could be rejected by global or endpoint rate limiting policies. Or the request could be cancelled while waiting for the lease.
+
+Meter name is `Microsoft.AspNetCore.RateLimiting`
+<!-- endsemconv -->
+
+<!-- semconv metric.aspnet.rate_limiting.requests(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `aspnet.rate_limiting.policy` | string | Rate limiting policy name. | `fixed`; `sliding`; `token` | Conditionally Required: [1] |
+| `aspnet.rate_limiting.result` | string | Rate-limiting result, shows whether lease was acquired or contains rejection reason | `acquired`; `request_canceled` | Required |
```

Comment:
Would it make sense to return `aspnet.rate_limiting.reject_reason` name, remove `acquired` from the list, and make the attribute conditionally required? Then lack of rejection reason means that the lease was acquired. This is just a small optimization for the happy case.

Reply:
I'd like to keep "acquired" to make it obvious what has happened